### ISSUE TITLE
Progress component

### ISSUE THREAD

**Issue by joshwooding** [jpmorganchase/salt-ds#1872]
```[tasklist]
### Checklist
- [x] Density (HD, MD, LD, TD)
- [x] Mode (Light, Dark)
- [ ] Fill (Primary, Secondary, Tertiary)
- [ ] State (Default, Hover, Selected/Active, Disabled, Read-only, Other)
- [ ] Controlled API
- [ ] UITK accessibility parity (screen reader + keyboard interactions)
- [x] QA component with designer
```
```[tasklist]
### Deliverables
- [ ] https://github.com/jpmorganchase/salt-ds/issues/2588
- [ ] https://github.com/jpmorganchase/salt-ds/issues/2586
```
```[tasklist]
### Features
- [x] Realign component to design language, and update design tokens (where necessary).
- [x] Remove indeterminate variant (from Specs/ code), and release determinate only.
- [x] Remove size concept and create single size across densities.
```

**Comment by mark-tate** [comment]
Will create a separate ticket for docs and deliver just the component with this ticket

**Comment by feliciackh** [comment]
@dplsek design signed off, waiting on final dev reviews
